### PR TITLE
feat: add comments partial

### PR DESCRIPTION
**Pull request by rickvanprim**
## What problem does this PR solve?

Lack of support for a comments section at the bottom of posts.

## Is this PR adding a new feature?

Introduces a `comments.html` partial that is included at the bottom of `single.html` and defaults to using the builtin Disqus template if `.Site.DisqusShortname` is set.

## Is this PR related to any issue or discussion?

Closes #32

## PR Checklist

- [x] I have verified that the code works as described/as intended.
- [ ] This change adds a social icon which has a permissive license to use it.
- [x] This change **does not** include any external library/resources.
- [x] This change **does not** include any unrelated scripts (e.g. bash and python scripts).
- [ ] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

### CODE REVIEW

**Comment by hugo-sid** [comment]
Thanks for the PR, @rickvanprim.

Your contribution is greatly appreciated.

Could you post some screenshots of the blog pages, after successful Discuss integration?

**Comment by rickvanprim** [comment]
Enabled:
image

Disabled (builtin Disqus template disables when address is `localhost`/`127.0.0.1`):
image

**Comment by hugo-sid** [comment]
Thanks for the screenshots. The feature looks great.

It's time to add some documentation for this feature.
Could you please add some sample configuration to `config.toml` file? Also request you to include a small comment about the feature.

For example you can add something like this to `config.toml`:

```toml
#DisqusShortname = ABCDXYZ # provide discuss shortname & uncomment the line to enable discuss on blog post pages
```

**Comment by hugo-sid** [comment]
How come the Discuss background matches the theme background color? Did you configure it somewhere?

**Comment by rickvanprim** [comment]
Added some documentation as requested.

Regarding the theming, I haven't done anything.  It seems to be largely automatic, except that it breaks when you toggle light/dark until the page refreshes.

After toggling to light:
image
After refreshing:
image

Thoughts?

**Comment by hugo-sid** [comment]
Hi James,

>Added some documentation as requested.

Thanks. Could you push the change/commit? As of now, I only see the commit that you created 8 hours ago.

>Regarding the theming, I haven't done anything. It seems to be largely automatic, except that it breaks when you toggle light/dark until the page refreshes.

Thanks for the details. I have to find out how to fix this. We can address it later. Let's consider this an an enhancement. 
Your PR does a great job of adding support for comments.

File: README.md
Comment (hugo-sid) [inline README.md:205]: Great! Thanks for adding documentation in README.md file.

File: exampleSite/config.toml
Comment (hugo-sid) [inline exampleSite/config.toml:13]: You really have an 'eye for details' 😃. 
Thanks for the fix.

File: exampleSite/config.toml
Comment (hugo-sid) [inline exampleSite/config.toml:22]: Thanks for the simple, no-brainer comments/docs.

File: CHANGELOG.md
Comment (hugo-sid) [inline CHANGELOG.md:1]: This changelong is auto generated. I have setup a GitHub action workflow which:

- analyses the commits (from last release)
- determines the version to bump (it's easy since this project uses conventional commit specification)
- updates the changelog
- creates a git tag corresponding to the version

Finally it creates a PR with the above changes.  So, there is no manual editing required.

Can you please leave this file unchanged, so that we leave it to the CI to generate it?

This is not at all your mistake. I know that you did it with an intention to help me/the project 🙂.
I think I made a mistake in not mentioning it anywhere. Probably, I should have mentioned it in `CONTRIBUTING.md`.

**Review by hugo-sid — CHANGES_REQUESTED** [review]
Hi James. I have requested a change regarding `CHANGELOG.md`. Please see the comments below.

I will merge this PR shortly after this is resolved.

File: exampleSite/config.toml
Comment (rickvanprim) [reply exampleSite/config.toml:22]: I just copy-pasted your Google Analytics one 😉

File: CHANGELOG.md
Comment (rickvanprim) [reply CHANGELOG.md:1]: All good.  I wasn't sure how you were getting stable commit hashes in there, so I assumed it was post-PR 🙂

**Review by hugo-sid — APPROVED** [review]
Thanks @rickvanprim, for adding Discuss comments feature to this theme.

I appreciate your contribution.

Have a good day. Happy blogging :)

File: CHANGELOG.md
Comment (hugo-sid) [reply CHANGELOG.md:1]: > I wasn't sure how you were getting stable commit hashes in there, so I assumed it was post-PR

I see.

I have updated `CONTRIBUTING.md`.

Don't forget to star the theme, if you liked it.

Thanks.

**Pull request merged**
